src/voice.ts

```diff
@@ -257,11 +283,11 @@ export class Voice extends Element {
   }
 
   // Preformats the voice by applying the voice's stave to each note.
-  preFormat() {
+  preFormat(): this {
     if (this.preFormatted) return this;
 
     this.tickables.forEach((tickable) => {
-      if (!tickable.getStave()) {
+      if (!tickable.getStave() && this.stave) {
```

Comment:
This will silently fail, maybe throw an exception in the beginning of `preFormat` if `stave` is not defined.

Reply:
done